tests/components/watergate/test_init.py

```diff
@@ -0,0 +1,81 @@
+"""Tests for the Watergate integration init module."""
+
+from unittest.mock import patch
+
+from homeassistant.components import webhook
+from homeassistant.components.watergate.const import DOMAIN
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.const import CONF_WEBHOOK_ID
+from homeassistant.core import HomeAssistant
+from homeassistant.setup import async_setup_component
+
+from .const import MOCK_CONFIG
+
+from tests.common import ANY
+from tests.typing import ClientSessionGenerator
+
+
+async def test_async_setup_entry(
+    hass: HomeAssistant, mock_entry, mock_watergate_client
+) -> None:
+    """Test setting up the Watergate integration."""
+    hass.config.components.add("network")
+    hass.config.internal_url = "http://hassio.local"
+    mock_entry.add_to_hass(hass)
+
+    with (
+        patch(
+            "homeassistant.helpers.network.get_url", return_value="http://hassio.local"
```

Comment:
I don' t remember, but doesn't this return a mocked URL by default?

Reply:
It is not required anymore. I am not using this method anymore, and I am configuring it a few lines earlier with the internal_url, so now we have beautiful integration tests, almost E2E :)